src/Control/Monad/Error/Hoist.hs

```diff
@@ -41,6 +42,12 @@ instance MonadError e' m ⇒ HoistError m (Either e) e e' where
 instance (m ~ n, MonadError e' m) ⇒ HoistError m (EitherT e n) e e' where
   hoistError f = eitherT (throwError . f) return
 
+instance MonadError e' m ⇒ HoistError m (ExceptT e Identity) e e' where
+  hoistError f = either (throwError . f) return . runExcept
+
+instance MonadError e' m ⇒ HoistError m (ExceptT e m) e e' where
```

Comment:
I think I tried that first but got an type error for some reason, but maybe it was caused by something else. I'll try again...